src/extensions/step_handlers/wiot_handler/src/wiot_handler_1.cpp

```diff
@@ -0,0 +1,353 @@
+    }
+
+    if (!IsValidUpdateTypeInfo(workflowHandle))
+    {
+        return ADUCResult(ADUC_Result_Failure, ADUC_ERC_SWUPDATE_HANDLER_DOWNLOAD_FAILURE_WRONG_UPDATE_VERSION);
+    }
+
+    const size_t fileCount = workflow_get_update_files_count(workflowHandle);
+    if (fileCount != 1)
+    {
+        // For v1, only 1 file is expected.
+        Log_Error("Incorrect file count: %u", fileCount);
+        return ADUCResult(ADUC_Result_Failure, ADUC_ERC_SWUPDATE_HANDLER_DOWNLOAD_FAILURE_WRONG_FILECOUNT);
+    }
+
+    //
+    // Check to see if the content is already installed.
+    //
+
+    if (IsInstalled(workflowData).ResultCode == ADUC_Result_IsInstalled_Installed)
+    {
+        return ADUCResult(ADUC_Result_Download_Skipped_UpdateAlreadyInstalled);
+    }
+
+    //
+    // Download each of the files in the manifest
+    //
+
+    for (size_t fileIndex = 0; fileIndex < fileCount; fileIndex++)
+    {
+        if (workflow_is_cancel_requested(workflowHandle))
+        {
+            return Cancel(workflowData);
+        }
+
+        Log_Info("Downloading file #%u", fileIndex);
+
+        ADUC_FileEntity* entity_temp;
+        if (!workflow_get_update_file(workflowHandle, fileIndex, &entity_temp))
+        {
+            return ADUCResult(ADUC_Result_Failure, ADUC_ERC_SWUPDATE_HANDLER_DOWNLOAD_FAILURE_GET_PAYLOAD_FILE_ENTITY);
+        }
+        workflow_file_entity_ptr entity(entity_temp);
+
+        try
+        {
+            ADUCResult result = ExtensionManager::Download(
+                entity.get(),
+                workflowHandle,
+                &Default_ExtensionManager_Download_Options,
+                nullptr /*downloadProgressCallback*/);
+            if (result.IsResultCodeFailure())
+            {
+                Log_Error("Cannot download payload file#%u, error 0x%X", fileIndex, result.ExtendedResultCode());
+                return result;
+            }
+        }
+        catch (...)
+        {
+            return ADUCResult(
+                ADUC_Result_Failure, ADUC_ERC_SWUPDATE_HANDLER_DOWNLOAD_PAYLOAD_FILE_FAILURE_UNKNOWNEXCEPTION);
+        }
+    }
+
+    // return ADUCResult(ADUC_Result_Download_Success);
+    // TODO(JeffMill): TESTING -- return download failure for now.
+    return ADUCResult(ADUC_Result_Failure, ADUC_ERC_SWUPDATE_HANDLER_DOWNLOAD_FAILURE_GET_PAYLOAD_FILE_ENTITY);
+}
+
+/**
+ * @brief Install implementation
+ *
+ * @return ADUC_Result The result of the install.
+ * ADUC_Result_Install_Success - success
+ */
+ADUC_Result WiotHandler1::Install(const ADUC_WorkflowData* workflowData)
+{
+    Log_Info(HANDLER_LOG_ID "Install");
+
+    const ADUC_WorkflowHandle workflowHandle{ workflowData->WorkflowHandle };
+
+    if (workflow_is_cancel_requested(workflowHandle))
+    {
+        return Cancel(workflowData);
+    }
+
+    if (!IsValidUpdateTypeInfo(workflowHandle))
+    {
+        return ADUCResult(ADUC_Result_Failure, ADUC_ERC_UPDATE_CONTENT_HANDLER_INSTALL_FAILURE_UNKNOWNEXCEPTION);
```

Comment:
later, might want to create more granular failure ERCs for all error paths in this fn